packages/agreement-readmodel-writer/test/agreement-readmodel-writer.V1.integration.test.ts

```diff
@@ -0,0 +1,263 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { generateMock } from "@anatine/zod-mock";
+import {
+  getMockAgreement,
+  writeInReadmodel,
+} from "pagopa-interop-commons-test";
+import {
+  AgreementAddedV1,
+  AgreementConsumerDocumentAddedV1,
+  AgreementConsumerDocumentRemovedV1,
+  AgreementContractAddedV1,
+  AgreementDeletedV1,
+  AgreementDocument,
+  AgreementEventEnvelope,
+  AgreementStateV1,
+  AgreementUpdatedV1,
+  generateId,
+  toReadModelAgreement,
+} from "pagopa-interop-models";
+import { describe, expect, it } from "vitest";
+import { handleMessageV1 } from "../src/consumerServiceV1.js";
+import { toAgreementDocumentV1 } from "./protobufConverterToV1.js";
+import { agreements } from "./utils.js";
+
+describe("events V1", async () => {
+  it("should create an agreement", async () => {
+    const id = generateId();
+    const newAgreement: AgreementAddedV1 = {
+      agreement: {
+        id,
+        eserviceId: generateId(),
+        descriptorId: generateId(),
+        producerId: generateId(),
+        consumerId: generateId(),
+        state: AgreementStateV1.ACTIVE,
+        certifiedAttributes: [],
+        declaredAttributes: [],
+        verifiedAttributes: [],
+        createdAt: BigInt(new Date().getTime()),
+        consumerDocuments: [],
+      },
+    };
+    const message: AgreementEventEnvelope = {
+      event_version: 1,
+      sequence_num: 1,
+      stream_id: id,
+      version: 1,
+      type: "AgreementAdded",
+      data: newAgreement,
+      log_date: new Date(),
+    };
+    await handleMessageV1(message, agreements);
+
+    const agreement = await agreements.findOne({
+      "data.id": id.toString(),
+    });
+
+    expect(agreement?.data).toMatchObject({
+      id: newAgreement.agreement?.id,
+      eserviceId: newAgreement.agreement?.eserviceId,
+      descriptorId: newAgreement.agreement?.descriptorId,
+      producerId: newAgreement.agreement?.producerId,
+      consumerId: newAgreement.agreement?.consumerId,
+    });
```

Comment:
```suggestion
    expect(agreement?.data).toEqual(
      toReadModelAgreement(fromAgreementV1(newAgreement.agreement!))
    );
```
You can compare the whole object like this, I used the `toEqual` which ingnores the `undefined` properties.

You can make this change in all occurences where you need to compare the whole objects